internal/backend/factory/factory.go

```diff
@@ -81,29 +64,29 @@ func (handler *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	params, err := parseRequest(r, handler.state, handler.config)
-	if err != nil {
-		if errors.Is(err, errNotFound) {
-			httpNotFound(w)
-
-			return
-		}
+	path := r.URL.Path
 
-		handler.handleError("failed to parse request", w, err)
+	proxyURL, err := getProxyURL(handler.config.ImageFactoryBaseURL, path)
+	if err != nil {
+		handler.handleError("failed to get proxy URL", w, err)
 
 		return
 	}
 
-	handler.logger.Info("proxy request", zap.String("url", params.ProxyURL))
+	handler.logger.Info("proxy request", zap.String("url", proxyURL.String()))
 
-	proxyReq, err := http.NewRequestWithContext(r.Context(), r.Method, params.ProxyURL, nil)
+	proxyReq, err := http.NewRequestWithContext(r.Context(), r.Method, proxyURL.String(), nil)
 	if err != nil {
 		handler.handleError("failed to call image factory", w, err)
 
 		return
 	}
 
-	setContentHeaders(w, params.ContentType, params.DestinationFilename)
+	segments := strings.Split(path, "/")
+	filename := fmt.Sprintf("omni-%s-%s", config.Config.Account.Name, segments[len(segments)-1])
```

Comment:
For easier parsing, i changed it from `<platform>-<board>-<arch>-omni-<account-name>-<talos-version>.<ext>` to `omni-<account-name>-<platform>-<board>-<arch>.<ext>` but if necessary can try make it like it was before, or at least just include talos version.

I also dropped content-type which now defaults to `application/octet-stream`.

The reason for both of these is to not depend on the installation media list